recipes/repository.rb

```diff
@@ -106,36 +129,51 @@
   end
 
   # Add YUM repository
+  yumrepo_gpgkeys = []
+  if agent_major_version < 7
+    yumrepo_gpgkeys.push(node['datadog']['yumrepo_gpgkey'])
+  else
+    rpm_gpg_keys.each do |rpm_gpg_key|
+      yumrepo_gpgkeys.push(node['datadog']["yumrepo_gpgkey_new_#{rpm_gpg_key[rpm_gpg_keys_short_fingerprint]}"])
+    end
+  end
+
   yum_repository 'datadog' do
     description 'datadog'
     baseurl baseurl
     proxy node['datadog']['yumrepo_proxy']
     proxy_username node['datadog']['yumrepo_proxy_username']
     proxy_password node['datadog']['yumrepo_proxy_password']
-    gpgkey agent_major_version < 7 ? node['datadog']['yumrepo_gpgkey'] : node['datadog']['yumrepo_gpgkey_new']
+    gpgkey yumrepo_gpgkeys
     gpgcheck true
     action :create
   end
 when 'suse'
   # Import new RPM key
-  if node['datadog']['yumrepo_gpgkey_new']
+  rpm_gpg_keys.each do |rpm_gpg_key|
+    next unless node['datadog']["yumrepo_gpgkey_new_#{rpm_gpg_key[rpm_gpg_keys_short_fingerprint]}"]
+
     # Download new RPM key
-    new_key_local_path = ::File.join(Chef::Config[:file_cache_path], 'DATADOG_RPM_KEY_E09422B3.public')
-    remote_file 'DATADOG_RPM_KEY_E09422B3.public' do
+    new_key_local_path = ::File.join(Chef::Config[:file_cache_path], rpm_gpg_key[rpm_gpg_keys_name])
+    remote_file "remote_file_#{rpm_gpg_key[rpm_gpg_keys_name]}" do
```

Comment:
Is there a specific reason why the `remote_file_` prefix had to be added?

Reply:
It was to workaround this issue: https://app.circleci.com/pipelines/github/DataDog/chef-datadog/212/workflows/d4a699bf-7207-4004-bf2f-1ab269977678/jobs/2676

https://github.com/Foodcritic/foodcritic/issues/436